LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -0,0 +1,1384 @@
+                case VLCState.Stopped:
+                case VLCState.Ended:
+                case VLCState.NothingSpecial:
+                    UpdateSeekBar(TimeSpan.Zero);
+                    UpdatePauseAvailability(true);
+                    UpdateSeekAvailability(false);
+                    UpdateTracksSelectionButtonAvailability(AudioTracksSelectionButton, AudioSelectionUnavailableState);
+                    UpdateTracksSelectionButtonAvailability(ClosedCaptionsSelectionButton, ClosedCaptionsSelectionUnavailableState);
+                    goto case VLCState.Paused;
+                case VLCState.Paused:
+                    playPauseState = PlayState;
+                    break;
+                default:
+                    ShowError(null);
+                    playPauseState = PauseState;
+                    break;
+            }
+
+            Device.BeginInvokeOnMainThread(() =>
+            {
+                _ = FadeInAsync();
+                BufferingProgress = 0;
+                var playPauseButton = PlayPauseButton;
+                if (playPauseButton != null)
+                {
+                    VisualStateManager.GoToState(playPauseButton, playPauseState);
+                }
+            });
+            UpdateKeepScreenOn();
+        }
+
+        private void FadeOut()
+        {
+            Device.BeginInvokeOnMainThread(async () =>
+            {
+                if (await ControlsPanel.FadeTo(0, 1000) && ControlsPanel.Opacity == 0)
+                {
+                    ControlsPanel.IsVisible = false;
+                }
+            });
+        }
+
+        /// <summary>
+        /// Show an error message box.
+        /// </summary>
+        /// <param name="ex">The exception to show.</param>
+        protected virtual void ShowErrorMessageBox(Exception ex)
+        {
+            var error = ResourceManager.GetString(nameof(Strings.Error));
+            Device.BeginInvokeOnMainThread(() => this.FindAncestor<Page>().DisplayAlert(error, ex?.GetBaseException().Message ?? error,
+                ResourceManager.GetString(nameof(Strings.OK))));
+        }
+
+        /// <summary>
+        /// Controls fade in.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous operation.</returns>
+        public async Task FadeInAsync()
+        {
+            var controlsPanel = ControlsPanel;
+            if (controlsPanel == null)
+            {
+                return;
+            }
+
+            FadeOutTimer.Change(TimeSpan.FromMilliseconds(-1), TimeSpan.FromMilliseconds(-1));
+            controlsPanel.IsVisible = true;
+            if (controlsPanel.Opacity != 1)
+            {
+                await controlsPanel.FadeTo(1);
+            }
+            if (FadeOutEnabled && ShowAndHideAutomatically && MediaPlayer?.State == VLCState.Playing)
+            {
+                FadeOutTimer.Change(TimeSpan.FromSeconds(3), TimeSpan.FromMilliseconds(-1));
+            }
+        }
+    }
+}
```

Comment:
Can we reduce the file size/move things outside of this file?

Reply:
Is that really necessary? It's fine for me, mostly public properties and short private handlers.

Reply:
things are getting a bit complicated starting with `OnApplyTemplate` at line 602. I really think that this component is too big to be maintained efficiently, and it only has 3 buttons yet...